studybuddies/src/app/components/login/login.component.ts

```diff
@@ -7,9 +11,50 @@ import { Component, OnInit } from '@angular/core';
 })
 export class LoginComponent implements OnInit {
 
-  constructor() { }
+  constructor(
+    private authService: AuthService,
+    private router: Router,
+    private tokenStorage: TokenStorageService
+  ) 
+  {}
+
+  username = '';
+  password = '';
+  isLoggedIn = false;
+  isLoginFailed = false;
+  role = '';
 
   ngOnInit(): void {
+    if(this.tokenStorage.getUser()) {
+      this.isLoggedIn = true;
+      this.role = this.tokenStorage.getUser().role;
+    }
+    console.log(this.isLoggedIn)
+  }
+
+  login(){
+    if(!this.username) {
+      Swal.fire('Login', 'El nombre de usuario no es correcto', 'error')
+    }
+    if(!this.password) {
+      Swal.fire('Login', 'La contraseña indicada no es correcta', 'error')
+    }
+
+    this.authService.login(this.username, this.password).subscribe(response => {
+      this.tokenStorage.saveUser(response);
+      this.isLoginFailed = false;
+      this.isLoggedIn = true;
+      this.role = response.role
+      this.router.navigate(['/'])
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Team-Studybuddies/Studybuddies-Frontend/pullRequest?prid=7203036)
```suggestion
      this.router.navigate(["/"])
```